tests/unit-tests/graphpartitioning/test_metadata.py

```diff
@@ -0,0 +1,731 @@
+    file_metadata.file_prefix = "node_feature_"
+
+    with tempfile.TemporaryDirectory() as root_dir:
+        file_metadata.root_dir = root_dir
+        update_schema(input_dict, file_metadata)
+        run_test(root_dir, input_dict)
+
+    input_dict = {}
+    file_metadata.root_dir = os.getcwd()
+    update_schema(input_dict, file_metadata)
+    run_test(None, input_dict)
+
+
+@pytest.mark.parametrize("etypes", [[], [EDGE1], [EDGE1, EDGE2]])
+@pytest.mark.parametrize("file_type", ["csv", "numpy"])
+@pytest.mark.parametrize("delimiter", [" ", "/"])
+@pytest.mark.parametrize("create_files", [True, False])
+@pytest.mark.parametrize("prefix_dir", ["", "edges"])
+@pytest.mark.parametrize("is_absolute", [True, False])
+def test_edge_features(
+    etypes, file_type, delimiter, create_files, prefix_dir, is_absolute
+):
+    """Unit tests when edge features are present in the input graph."""
+    ntypes = [NODE1, NODE2]
+    input_dict = {}
+    with tempfile.TemporaryDirectory() as root_dir:
+
+        file_metadata = namedtuple("file_metadata", "")
+        file_metadata.root_dir = root_dir
+        file_metadata.prefix_dir = prefix_dir
+        file_metadata.create_files = create_files
+        file_metadata.is_absolute = is_absolute
+        file_metadata.file_prefix = "edge_feature_"
+
+        add_nodes(input_dict, {ntype: 10 for ntype in ntypes})
+        add_edge_features(
+            input_dict,
+            {etype: 10 for etype in etypes},
+            {etype: [EFEAT1] for etype in etypes},
+            {etype: 5 for etype in etypes},
+            {
+                etype + "/" + EFEAT1: {NAME: file_type, DELIMITER: delimiter}
+                for etype in etypes
+            },
+            file_metadata,
+        )
+
+        filename = os.path.join(root_dir, METADATA_NAME)
+        with open(filename, "w") as input_handle:
+            json.dump(input_dict, input_handle, indent=4)
+
+        obj = MetadataSchema(METADATA_NAME, root_dir, init_maps=False)
+        obj._init_etype_maps()
+
+        try:
+            obj._init_etype_feature_files()
+            if create_files:
+                for idx, etype in enumerate(etypes):
+                    (
+                        actual_file_type,
+                        actual_delimiter,
+                        data_files,
+                    ) = obj.get_etype_feature_files(etype, EFEAT1)
+                    assert (
+                        file_type == actual_file_type
+                    ), f"File type does not match."
+                    assert (
+                        delimiter == actual_delimiter
+                    ), f"Delimiters does not match."
+                    assert (
+                        len(data_files) == 5
+                    ), f"No. of edge feature files does not match."
+        except AssertionError as exp:
+            if create_files:
+                raise exp
+            else:
+                pass
+
+
+@pytest.mark.parametrize(
+    "metadata",
```

Comment:
this is supercomplex.

Reply:
Removed as this has already been tested by other test cases.